### PR TITLE
fix(pci.private-network): reload states when redirecting

### PR DESCRIPTION
**Pull request by marie-j**
MSB-119

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/private-networks/vrack/vrack.routing.js [hunk 1]
```diff
@@ -1,12 +1,18 @@
+import isEmpty from 'lodash/isEmpty';
+
 export default /* @ngInject */($stateProvider) => {
   $stateProvider.state('pci.projects.project.privateNetwork.vrack', {
     url: '/onboarding',
     component: 'pciProjectPrivateNetworksVrack',
     resolve: {
       createVrack: ($state, projectId) => () => $state.go('pci.projects.project.privateNetwork.vrack.new', { projectId }),
-      onVrackCreated: ($state, projectId) => () => $state.go('pci.projects.project.privateNetwork', { projectId }),
+      onVrackCreated: ($state, projectId) => () => $state.go('pci.projects.project.privateNetwork', { projectId }, { reload: true }),
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/private-networks/vrack/vrack.routing.js]: Both resolve methods could have explicit annotation `/* @ngInject */`.

File: packages/manager/modules/pci/src/projects/project/private-networks/vrack/add/add.routing.js [hunk 2]
```diff
@@ -6,8 +6,8 @@ export default /* @ngInject */($stateProvider) => {
     resolve: {
       goBack: ($state, projectId) => () => $state.go('^', { projectId }),
       messageContainer: () => 'pci.projects.project.privateNetwork.vrack',
-      onCreationSuccess: ($state, projectId) => () => $state.go('pci.projects.project.privateNetwork.vrack', { projectId }),
-      onAssociationSuccess: ($state, projectId) => () => $state.go('pci.projects.project.privateNetwork', { projectId }),
+      onCreationSuccess: ($state, projectId) => () => $state.go('pci.projects.project.privateNetwork.vrack', { projectId }, { reload: true }),
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/private-networks/vrack/add/add.routing.js]: Resolve methods could have explicit annotation `/* @ngInject */`.

File: packages/manager/modules/pci/src/projects/project/private-networks/vrack/add/add.service.js [hunk 3]
```diff
@@ -24,6 +24,16 @@ export default class {
           status: 'done',
         },
       },
-    );
+    )
+      .then(
```
Comment (FredericEspiau) [inline packages/manager/modules/pci/src/projects/project/private-networks/vrack/add/add.service.js]:
```suggestion
      .then(() => true)
      .catch((error) => 
          error.status === 404
              ? true
              : error
      )
```
?

Comment (marie-j) [reply packages/manager/modules/pci/src/projects/project/private-networks/vrack/add/add.service.js]: Rejection is needed when we have another error than 404 to be catched by transitions hook and go to error page

Comment (FredericEspiau) [reply packages/manager/modules/pci/src/projects/project/private-networks/vrack/add/add.service.js]: true ! But the calls to `Promise.resolve` could be removed, couldn't they ?

**Comment by antleblanc** [comment]
@marie-j Is there any internal ticket reference related to this pull request?

Thanks!

**Comment by marie-j** [comment]
> @marie-j Is there any internal ticket reference related to this pull request?
> 
> Thanks!

@antleblanc Description updated

**Comment by antleblanc** [comment]
Rebased on `master` (81941ef1a7d54665d14de3a594a9fc76f5b0af15)

**Pull request merged**